lib/ash_authentication/strategies/password/sign_in_preparation.ex

```diff
@@ -25,8 +25,11 @@ defmodule AshAuthentication.Strategy.Password.SignInPreparation do
     identity_field = strategy.identity_field
     identity = Query.get_argument(query, identity_field)
 
-    query
-    |> Query.filter(^ref(identity_field) == ^identity)
+    if identity == nil do
+      query |> Query.filter(is_nil(^ref(identity_field)))
```

Comment:
You should not be able to sign in as a user that has no email or username. Is this fix for an error that you found?

Reply:
No, I just want to remove following warning.


```
2024-08-11T22:24:55Z app[e2867659f46678] nrt [info]warning: Comparing values with `nil` will always return `false`. Use `is_nil/1` instead. In: `{:_ref, [], :email} == nil`
2024-08-11T22:24:55Z app[e2867659f46678] nrt [info]  (ash_authentication 4.0.2) lib/ash_authentication/strategies/password/sign_in_preparation.ex:29: AshAuthentication.Strategy.Password.SignInPreparation.prepare/3
2024-08-11T22:24:55Z app[e2867659f46678] nrt [info]  (ash 3.3.3) lib/ash/query/query.ex:708: anonymous fn/6 in Ash.Query.run_preparations/6
2024-08-11T22:24:55Z app[e2867659f46678] nrt [info]  (elixir 1.17.1) lib/enum.ex:4858: Enumerable.List.reduce/3
2024-08-11T22:24:55Z app[e2867659f46678] nrt [info]  (elixir 1.17.1) lib/enum.ex:2585: Enum.reduce_while/3
2024-08-11T22:24:55Z app[e2867659f46678] nrt [info]  (ash 3.3.3) lib/ash/query/query.ex:552: Ash.Query.for_read/4
2024-08-11T22:24:55Z app[e2867659f46678] nrt [info]  (ash_phoenix 2.1.1) lib/ash_phoenix/form/form.ex:878: AshPhoenix.Form.for_read/3
```